cpp/src/Ice/OutgoingAsync.cpp

```diff
@@ -559,7 +559,14 @@ OutgoingAsync::OutgoingAsync(ObjectPrx proxy, bool synchronous)
 void
 OutgoingAsync::prepare(string_view operation, OperationMode mode, const Context& context)
 {
-    checkSupportedProtocol(_proxy._getReference()->getProtocol());
+    if (_proxy._getReference()->getProtocol().major != currentProtocol.major)
```

Comment:
I kept and sometime restored the logic if the protocol is 1.x, we automatically downgrade to 1.0 when making an invocation.

Obviously, none of this ever happens.